packages/hydrogen/src/Numbers/useNumberValue.tsx

```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+import BigNumber from 'bignumber.js';
+import { bigNumToNumber, UINumberDecimalAspect } from './runOnUILocalizedNumberFormat';
+
+export function useNumberValue(rawValue: number | BigNumber, decimalAspect: UINumberDecimalAspect) {
+    return React.useMemo(
+        () => (rawValue instanceof BigNumber ? bigNumToNumber(rawValue, decimalAspect) : rawValue),
+        [rawValue],
+    );
```

Comment:
Please, think of changes suggested above (i.e. to split the rawValue into two parts and operate them safely)